pkg/cache/sql/integration_test.go

```diff
@@ -0,0 +1,324 @@
+					Partial: false,
+				},
+			),
+			wantNames: []string{"matches-filter", "partial-matches", "not-matches-filter"},
+		},
+		{
+			name: "or filters on different fields",
+			filters: orFiltersForFilters(
+				informer.Filter{
+					Field:   []string{`metadata`, `annotations[somekey]`},
+					Match:   "somevalue",
+					Op:      informer.Eq,
+					Partial: true,
+				},
+				informer.Filter{
+					Field:   []string{`metadata`, `name`},
+					Match:   "missing",
+					Op:      informer.Eq,
+					Partial: false,
+				},
+			),
+			wantNames: []string{"matches-filter", "partial-matches", "missing"},
+		},
+		{
+			name: "and filters, both must match",
+			filters: []informer.OrFilter{
+				{
+					Filters: []informer.Filter{
+						{
+							Field:   []string{`metadata`, `annotations[somekey]`},
+							Match:   "somevalue",
+							Op:      informer.Eq,
+							Partial: true,
+						},
+					},
+				},
+				{
+					Filters: []informer.Filter{
+						{
+							Field:   []string{`metadata`, `name`},
+							Match:   "matches-filter",
+							Op:      informer.Eq,
+							Partial: false,
+						},
+					},
+				},
+			},
+			wantNames: []string{"matches-filter"},
+		},
+		{
+			name: "no matches",
+			filters: orFiltersForFilters(
+				informer.Filter{
+					Field:   []string{`metadata`, `annotations[somekey]`},
+					Match:   "valueNotRepresented",
+					Op:      informer.Eq,
+					Partial: false,
+				},
+			),
+			wantNames: []string{},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		i.Run(test.name, func() {
+			options := informer.ListOptions{
+				Filters: test.filters,
+			}
+			partitions := []partition.Partition{defaultPartition}
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+			cfgMaps, total, continueToken, err := cache.ListByOptions(ctx, options, partitions, testNamespace)
+			i.Require().NoError(err)
+			// cancel here so that we don't build up a large list of defer'd cancels
+			cancel()
+			i.Require().Equal(len(test.wantNames), total)
+			// since there's no additional pages, the continue token should be empty
+			i.Require().Equal("", continueToken)
+			i.Require().NotNil(cfgMaps)
+			i.Require().Len(cfgMaps.Items, len(test.wantNames))
+			requireNames := map[string]struct{}{}
+			for _, name := range test.wantNames {
+				requireNames[name] = struct{}{}
+			}
+			for _, configMap := range cfgMaps.Items {
+				i.Require().Contains(requireNames, configMap.GetName())
+				delete(requireNames, configMap.GetName())
+			}
+		})
+	}
+}
+
+func (i *IntegrationSuite) createCacheAndFactory(fields [][]string, transformFunc cache.TransformFunc) (*factory.Cache, *factory.CacheFactory, error) {
+	cacheFactory, err := factory.NewCacheFactory()
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to make factory: %w", err)
+	}
+	dynamicClient, err := dynamic.NewForConfig(&i.restCfg)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to make dynamicClient: %w", err)
```

Comment:
Instead of returning an error maybe this fails the test?

Reply:
For side functions like this I prefer to return an error. The main function that's actually running the test can make the require/fail assertions to actually fail the test.